File: quantum-web/quantum-web-user/src/main/java/com/lunarstra/quantum/model/converter/UserCoinConverter.java
package com.lunarstra.quantum.model.converter;

import com.lunarstra.quantum.model.entity.UserCoin;
import com.lunarstra.quantum.model.request.AddUserCoinRequest;
import com.lunarstra.quantum.model.request.UpdateUserCoinRequest;
import com.lunarstra.quantum.model.response.UserCoinResponse;

/**
 * 用户硬币 转换类。
 *
 * @author lunarstra
 * @since 2025-10-18
 */

public class UserCoinConverter {

    /**
     * entity -> response
     *
     * @return
     */
    public static UserCoinResponse entityConvert2Response(UserCoin userCoin) {
        if (userCoin == null) {
            return null;
        }
        UserCoinResponse userCoinResponse = new UserCoinResponse();

        userCoinResponse.setUserId(userCoin.getUserId());
        userCoinResponse.setCoinNum(userCoin.getCoinNum());
        userCoinResponse.setCreateTime(userCoin.getCreateTime());
        userCoinResponse.setUpdateTime(userCoin.getUpdateTime());

        return userCoinResponse;
    }

    /**
     * addRequest -> entity
     *
     * @return
     */
    public static UserCoin addRequestConvert2Entity(AddUserCoinRequest addUserCoinRequest) {
        if (addUserCoinRequest == null) {
            return null;
        }
        UserCoin userCoin = new UserCoin();

        userCoin.setUserId(addUserCoinRequest.getUserId());
        userCoin.setCoinNum(addUserCoinRequest.getCoinNum());

        return userCoin;
    }

    /**
     * updateRequest -> entity
     *
     * @return
     */
    public static UserCoin updateRequestConvert2Entity(UpdateUserCoinRequest updateUserCoinRequest) {
        if (updateUserCoinRequest == null) {
            return null;
        }
        UserCoin userCoin = new UserCoin();

        userCoin.setUserId(updateUserCoinRequest.getUserId());
        userCoin.setCoinNum(updateUserCoinRequest.getCoinNum());

        return userCoin;
    }
}

File: quantum-web/quantum-web-user/src/main/java/com/lunarstra/quantum/model/converter/UserCoinConverter.kt
package com.lunarstra.quantum.model.converter

import com.lunarstra.quantum.model.entity.UserCoin
import com.lunarstra.quantum.model.request.AddUserCoinRequest
import com.lunarstra.quantum.model.request.UpdateUserCoinRequest
import com.lunarstra.quantum.model.response.UserCoinResponse

/**
 * 用户硬币 转换类。
 *
 * @author lunarstra
 * @since 2025-10-18
 */
object UserCoinConverter {
    /**
     * entity -> response
     *
     * @return
     */
    @JvmStatic
    fun entityConvert2Response(userCoin: UserCoin?): UserCoinResponse? {
        if (userCoin == null) {
            return null
        }
        val userCoinResponse = UserCoinResponse().apply {

            userId = userCoin.userId
            coinNum = userCoin.coinNum
            createTime = userCoin.createTime
            updateTime = userCoin.updateTime
        }

        return userCoinResponse
    }

    /**
     * addRequest -> entity
     *
     * @return
     */
    @JvmStatic
    fun addRequestConvert2Entity(addUserCoinRequest: AddUserCoinRequest?): UserCoin? {
        if (addUserCoinRequest == null) {
            return null
        }
        val userCoin = UserCoin().apply {

            userId = addUserCoinRequest.userId
            coinNum = addUserCoinRequest.coinNum
        }

        return userCoin
    }

    /**
     * updateRequest -> entity
     *
     * @return
     */
    @JvmStatic
    fun updateRequestConvert2Entity(updateUserCoinRequest: UpdateUserCoinRequest?): UserCoin? {
        if (updateUserCoinRequest == null) {
            return null
        }
        val userCoin = UserCoin().apply {

            userId = updateUserCoinRequest.userId
            coinNum = updateUserCoinRequest.coinNum
        }

        return userCoin
    }
}